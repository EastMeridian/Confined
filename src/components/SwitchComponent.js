import React from 'react';
import t from 'prop-types';

const SwitchComponent = ({
  components,
  component,
  fallbackComponent,
  ...other
}) => {
  const Component = components[component] || fallbackComponent;
  return <Component {...other} component={component} />;
};

SwitchComponent.propTypes = {
  components: t.objectOf(t.elementType),
  component: t.string,
  fallbackComponent: t.node,
};

SwitchComponent.defaultProps = {
  components: {},
  component: 'div',
  fallbackComponent: 'div',
};

export default SwitchComponent;
