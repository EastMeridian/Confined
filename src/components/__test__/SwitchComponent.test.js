import '@testing-library/jest-dom';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';
import React from 'react';
import SwitchComponent from '../SwitchComponent';
import '@testing-library/jest-dom/extend-expect';

describe('SwitchComponent component', () => {
  it('should matches the snapshot', () => {
    const component = create(<SwitchComponent />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should display fallbackComponent if no component or components provided', () => {
    const { container } = render(<SwitchComponent />);
    expect(container.firstChild.nodeName).toBe('DIV');
  });

  it('should display Component from a component store', () => {
    const components = {
      test: () => <div>test div</div>,
    };

    const { getByText } = render(
      <SwitchComponent components={components} component="test" />,
    );
    expect(getByText('test div')).toBeInTheDocument();
  });
});
