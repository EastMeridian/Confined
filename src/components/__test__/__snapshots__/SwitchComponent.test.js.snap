// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchComponent component should matches the snapshot 1`] = `
<div
  component="div"
/>
`;
